src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -36,11 +36,8 @@ public static bool IsTrackedResource(Schema schema, Dictionary<string, Schema> d
         }
 
         // determine if the operation is xms pageable or returns an object of array type
-        public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDefinition entity)
+        public static bool IsArrayTypeResponseOperation(Operation op, ServiceDefinition entity)
```

Comment:
@sarangan12 just did a very similar thing, once he merges, you should share this. https://github.com/Azure/autorest/pull/2002/files